agent-inject/agent/annotations.go

```diff
@@ -812,6 +826,15 @@ func (a *Agent) templateConfigExitOnRetryFailure() (bool, error) {
 	return strconv.ParseBool(raw)
 }
 
+func (a *Agent) templateConfigMaxConnectionsPerHost() (int64, error) {
+	raw, ok := a.Annotations[AnnotationTemplateConfigMaxConnectionsPerHost]
+	if !ok {
+		return DefaultTemplateConfigMaxConnectionsPerHost, nil
+	}
+
+	return strconv.ParseInt(raw, 10, 64)
```

Comment:
Would `parseutil.ParseInt` be better here?

Reply:
I know we try and use it where we can, but I don't mind if we think it's overkill

Reply:
I was on the fence about that and chose to remain consistent with the current implementation. I like the improvement, however, and will make the change. Thanks!